### PR TITLE
Start adding pipeline to collector

### LINKED ISSUE #169: cmd/occollector: support for receivers.
Acceptance criteria:
* Supports interceptors for Zipkin and Jaeger
* Support for config file
* Test coverage
* Update README as applicable

### PR DESCRIPTION
**Pull request by pjanotti**
This is the basis to do the work for actually implementing OC collector
in the short run. At first enable receivers:

1. OC Receiver
2. Jaeger
3. Zipkin

The queue processor is a no-op at this stage and should be the next step
added. As previously indicated before, there is some duplication of code
with the agent, this should be reduced as we implement the interfaces and
project coalesce around certain patterns - in this regard there is clearly
temporary code here (e.g.: the interface StartTraceReceiver/StopTraceReceiver
can help deal with some redundant code).

In order to avoid a messy merger regarding the Jaeger receiver I dropped the
temp one that I was using and lifted the one from @odeke-em PR #192 (the merge
shouldn't be hard in any order).

Fix #169

### CODE REVIEW

File: internal/collector/processor/processor.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import (
+	"go.uber.org/zap"
+
+	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
+)
+
+// SpanProcessor handles batches of spans converted to OpenCensus proto format.
+type SpanProcessor interface {
+	// ProcessSpans processes spans and return with either a list of true/false success or an error
+	ProcessSpans(batch *agenttracepb.ExportTraceServiceRequest, spanFormat string) ([]bool, error)
```
Comment (pjanotti) [inline internal/collector/processor/processor.go]: The []bool perhaps is overkill and simply acknowledgement numbers are enough (or perhaps a []int with the failures indexes, nil for typical case: all spans ok). However, for now let's keep as with initial definition until we have the full pipeline and then we make our decision.

**Comment by odeke-em** [comment]
Hello @pjanotti, since we've merged the Jaeger receiver code in https://github.com/census-instrumentation/opencensus-service/pull/192 please rebase and we can get started with this review.

File: cmd/occollector/app/collector/collector.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	"log"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/census-instrumentation/opencensus-service/internal/collector/jaeger"
+	"github.com/census-instrumentation/opencensus-service/internal/collector/opencensus"
+	"github.com/census-instrumentation/opencensus-service/internal/collector/processor"
+	"github.com/census-instrumentation/opencensus-service/internal/collector/zipkin"
```
Comment (odeke-em) [inline cmd/occollector/app/collector/collector.go]: It'd be nice to separate the OpenCensus imports from the others i.e. please place the others before the opencensus* imports
```go
   "github.com/spf13/cobra"
   "github.com/spf13/viper"
   "go.uber.org/zap"
   "go.uber.org/zap/zapcore"

   "github.com/census-instrumentation/opencensus-service/internal/collector/jaeger"
   ....
)
```

**Comment by pjanotti** [comment]
I will close this one and open another one later today that already adds the queue processor and some more tests, it will be larger but brings us closer to our goals for the sprint.

**Pull request closed without merging**